src/symmetry.jl

```diff
@@ -66,13 +66,32 @@ end
 Filter out the symmetry operations that respect the symmetries of the discrete BZ grid
 """
 function symmetries_preserving_kgrid(symmetries, kcoords)
+    kcoords_normalized = normalize_kpoint_coordinate.(kcoords)
+    T = eltype(kcoords[1])
+    tol = T <: Rational ? 0 : sqrt(eps(T))
+    is_approx_in(x, X, tol) = any(y -> norm(x-y) ≤ tol, X)
     function preserves_grid(S)
-        all(normalize_kpoint_coordinate(S * k) in kcoords
-            for k in normalize_kpoint_coordinate.(kcoords))
+        all(is_approx_in(normalize_kpoint_coordinate(S * k), kcoords_normalized, tol)
+            for k in kcoords_normalized)
     end
     filter(symop -> preserves_grid(symop[1]), symmetries)
 end
 
+
+"""
+Find the subset of symmetries compatible with the grid induced by the given kcoords and ksymops
+"""
+function symmetries_preserving_kcoords_ksymops(symmetries, kcoords, ksymops)
```

Comment:
I wonder if it would be nicer/easier to read if we expose `preserves_grid` as a function as well as a function to build all `kcords` (we might already have this) and then just perform the action of this function beast by just using `filter` and the latter two functions.

Reply:
I'd rather not anticipate possible future API needs that we don't know yet, so let's just do that if we find that we need preserves_grid. Hopefully we will never have to touch this code ever again (where have I heard that before?...)

Reply:
Hmm. I understand you don't want to touch this again, but in my experience it's better to build complex functionality like this by chaining small functions (such as preserve_kgrids predicate and a generator for all kcoords). A generator for all kcoords sounds like a pattern we might actually have somewhere in the code without realising it for other reasons (I have not checked though).